transforms/shared.js

```diff
@@ -1,35 +1,100 @@
-import { PATCHED_MATH_METHODS } from "../src/constants.js";
+import {
+  PATCHED_MATH_METHODS,
+  PATCHED_DECIMAL_METHODS,
+} from "../src/constants.js";
+
+const builtInLibraryName = "Decimal";
+
+const coerceConstructorArg = (path, t) => {
+  const args = path.get("arguments");
+  const [first] = args;
+
+  if (
+    !args.length ||
+    first.isNullLiteral() ||
+    first.isIdentifier({ name: "undefined" })
+  ) {
+    throw path.buildCodeFrameError(
+      new TypeError(`Can't convert null or undefined to Decimal.`)
+    );
+  }
+
+  if (first.isBooleanLiteral()) {
+    return t.NumericLiteral(Number(first.node.value));
+  }
+
+  if (first.isBigIntLiteral()) {
+    return t.StringLiteral(first.node.value);
+  }
+
+  return first.node;
+};
+
+const handleIdentifierCall = (
+  path,
+  implementationIdentifier,
+  knownDecimalNodes,
+  t
+) => {
+  if (path.get("callee").isIdentifier({ name: implementationIdentifier })) {
+    knownDecimalNodes.add(path.node);
+    return;
+  }
+
+  if (path.get("callee").isIdentifier({ name: builtInLibraryName })) {
+    const val = coerceConstructorArg(path, t);
+    const newNode = t.callExpression(t.Identifier(implementationIdentifier), [
+      val,
+    ]);
+
+    knownDecimalNodes.add(newNode);
+    path.replaceWith(newNode);
+    path.skip();
+    return;
+  }
+};
+
+const handleMemberCall = (path, knownDecimalNodes) => {
+  const callee = path.get("callee");
+  const object = callee.get("object");
+  const property = callee.get("property");
+  const methodName = property.isIdentifier()
+    ? property.node.name
+    : property.node.value;
+
+  if (
+    object.isIdentifier({ name: "Math" }) &&
+    PATCHED_MATH_METHODS.includes(methodName)
+  ) {
+    knownDecimalNodes.add(path.node);
+    return;
+  }
 
-export const addToDecimalNodes =
+  if (
+    object.isIdentifier({ name: builtInLibraryName }) &&
+    PATCHED_DECIMAL_METHODS.includes(methodName)
+  ) {
+    knownDecimalNodes.add(path.node);
+  }
+};
```

Comment:
I hate all these ifs but I guess that's a good reason to want the pattern-matching proposal. 😆

Reply:
If you wanted it to be more like a match statement, this refactor makes it easy for you to make it into `if (...) { ... } else if (...) { ... }`, instead of the early return idiom. But I don't have a problem with the way it is now.